src/mpid/ch4/src/ch4_impl.h

```diff
@@ -1052,15 +1052,16 @@ static inline int MPIDI_CH4U_destroy_shm_segment(MPI_Aint shm_segment_len,
 
 /* Compute accumulate operation.
  * The source datatype can be only predefined; the target datatype can be
- * predefined or derived. */
+ * predefined or derived. If the source buffer has been packed by the caller,
+ * src_packed must be set to 1.*/
 #undef FUNCNAME
 #define FUNCNAME MPIDI_CH4U_compute_acc_op
 #undef FCNAME
 #define FCNAME MPL_QUOTE(FUNCNAME)
 MPL_STATIC_INLINE_PREFIX int MPIDI_CH4U_compute_acc_op(void *source_buf, int source_count,
                                                        MPI_Datatype source_dtp, void *target_buf,
                                                        int target_count, MPI_Datatype target_dtp,
-                                                       MPI_Op acc_op)
+                                                       MPI_Op acc_op, int src_packed)
```

Comment:
Let's use `bool` and `true/false` for newly introduced variables.

Reply:
Good catch. Will fix.

Reply:
Or... define an enum
```C
{
     MPIDI_SRC_PACKED,
     MPIDI_SRC_UNPACKED,
}
```
then pass these values. That way we can avoid awkward comments like
```C
fun(..., 0 /* src_packed */);
```

Reply:
👍 I think enum is the generally preferred style for these types of arguments for the reason you show. Less interpretation for the reader.

Reply:
I am not sure if `UNPACKED` is suitable here, as the buffer is not "unpacked" from another buffer. The parameter is simply a TRUE/FALSE flag.

Reply:
Similer code I have seen in ch4 is 
```
enum {
    MPIDI_CH4R_NETMOD = 0,
    MPIDI_CH4R_SHM = 1
};
```
But this defines two types, not a true/false flag.

Reply:
Maybe `PACKED, PLAIN`.